documentation/design-docs/dotnet-tools.md

```diff
@@ -398,6 +399,45 @@ CONVERT
       Writing:       ./trace.speedscope.json
       Conversion complete
 
+REPORT
+
+    dotnet-trace report [-h|--help]
+                        [-t|--type]
+                        [-p|--parameters]
+                        [-o|--output <output_file_path>]
+                        --format <format>
+                        <trace_file>
+
+    Generate report into stdout from a previously generated trace_file.
+
+    -h, --help
+        Show command line help
+    
+    -t, --type
+        Specifies the type of report to create. The first type of report that will be available is 'topN'
+
+    -p, --parameters
+        Takes a list of arguments for each specific report type, see below for details.
+        type            parameters  explanation
+        'topN'          -n          the number of top functions to gather on the stack
+                        -i          find the top n methods inclusively on the stack. Without this tag, will find the top N methods exclusively on the stack
+        'countEvent'    
```

Comment:
@jonsequitur is it possible to arbitrarily nest command handlers? Specifically, could we do something like `tool verb verb [options]`? I'm imagining it may be easier to be able to have different reports be sub-commands off the `report` command, e.g.,

`dotnet trace report TopN -n 100 <trace file>`
`dotnet trace report eventstat --filter "MyProvider/*;Microsoft-*" <trace file>`

etc.

If that's the case, we may want to update `dotnet-gcdump report` the same way.

Reply:
You can do this by directly calling one handler from within another in your code.

Reply:
Does that mean the parent handler has to accept the union of all possible arguments to any sub-commands? For this example, would the `report` handler need to bind all possible arguments `TopN` and `CountEvents` might take, or does the binder know to bind the arguments again when the next command handler is called?

Reply:
System.CommandLine would be unaware of the call to the other command's handler because you'd just be passing the arguments directly, but the default binding will look up arguments defined on the parent. 

![image](https://user-images.githubusercontent.com/547415/136105598-39cb971b-4e0d-48cf-930a-c74054bb09f7.png)

You could additionally use model binding a custom class in both handlers to avoid repeating the parameter definitions.